tests/test_servo.py

```diff
@@ -307,30 +315,59 @@ def test_load_configuration_to_subnode_zero(read_config, pytestconfig, connect_t
 
 @pytest.mark.canopen
 @pytest.mark.ethernet
-def test_store_parameters(connect_to_slave):
+@pytest.mark.ethercat
+def test_store_parameters(connect_to_slave, request):
+    user_over_voltage_register = "DRV_PROT_USER_OVER_VOLT"
+
     servo, net = connect_to_slave
-    assert servo is not None and net is not None
+
+    initial_user_over_voltage_value = servo.read(user_over_voltage_register)
+    new_user_over_voltage_value = initial_user_over_voltage_value + 5
+
+    servo.write(user_over_voltage_register, new_user_over_voltage_value)
+
+    assert servo.read(user_over_voltage_register) == new_user_over_voltage_value
 
     servo.store_parameters()
 
-    value = servo.read("DRV_STATE_STATUS")
-    assert value is not None
+    time.sleep(5)
+
+    request.getfixturevalue("perform_power_cycle")
```

Comment:
I find this a bit confusing to read.
Why don't we request on the pytest function arguments the rack service client, and then request the power cycle with the client handle there?
Although this pattern appears to be working I think fixtures are not meant to be used like that and at least I'm aware of a potential problem: Fixtures can, in some cases, be internally cached by pytest.
https://docs.pytest.org/en/stable/how-to/fixtures.html#fixtures-can-be-requested-more-than-once-per-test-return-values-are-cached

Reply:
I did it this way so we can reuse the same fixture on different tests. 
Yes, you're right it's cached if the fixture is requested more than once on the same test. 
I just don't see a test that will need to call this fixture more than once.